modin/core/execution/dispatching/factories/dispatcher.py

```diff
@@ -109,7 +109,8 @@ class FactoryDispatcher(object):
     @classmethod
     def get_factory(cls) -> factories.BaseFactory:
         """Get current factory."""
-        # mostly for testing
+        Engine.subscribe(cls._update_factory)
+        StorageFormat.subscribe(cls._update_factory)
```

Comment:
This would subscribe multiple times, while originally it was one callback per module import (hence one callback per lifetime); maybe `Parameter.subscribe()` should be modified to only add a callback once? As now it just adds them to a list to maintain a FIFO order.

Reply:
I think it's cleaner to make sure we subscribe once. I did that.